### PR TITLE
Feature/7221 update get order summary logic to set complete for OrderParty

### PR DESCRIPTION
**Pull request by SimonLLawrence**

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs [hunk 1]
```diff
@@ -144,6 +144,242 @@ public async Task GetOrderSummaryAsync_IsSummaryComplete_ReturnResult()
             orderSummary.Should().BeEquivalentTo(expected);
         }
 
+        [Test]
+        public async Task GetOrderSummaryAsync_OrderPartyOrganisationNameMissing_ReturnsOrderPartyIncomplete()
+        {
+            const string orderId = "C0000014-01";
+            var context = OrdersControllerTestContext.Setup();
+
+            (Order order, OrderSummaryModel expected) = CreateOrderSummaryTestData(orderId, "Some Description", context.PrimaryOrganisationId);
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs:153]: A lot of these tests are the same. I wonder if we can utilise `TestCaseSource`. See example found in ApplicationUserValidatorTests in Identitiy.Api or maybe @bjss-chris has a good idea?

Comment (bjss-chris) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs:153]: One option might be to use an expression tree, like this for example:

```c#
[Test]
public async Task GetOrderSummaryAsync_OrderPartyOrganisationNameMissing_ReturnsOrderPartyIncomplete()
{
	await GetOrderSummaryAsync_PropertyMissing_ReturnsOrderPartyIncomplete(o => o.OrganisationName);
}

private static async Task GetOrderSummaryAsync_PropertyMissing_ReturnsOrderPartyIncomplete<T>(Expression<Func<Order, T>> selector)
{
	const string orderId = "C0000014-01";
	var context = OrdersControllerTestContext.Setup();

	(Order order, OrderSummaryModel expected) = CreateOrderSummaryTestData(orderId, "Some Description", context.PrimaryOrganisationId);

	var orderParameter = selector.Parameters.First();
	var assignDefaultExpression = Expression.Lambda<Action<Order>>(
		Expression.Assign(selector.Body, Expression.Constant(default(T), typeof(T))), orderParameter);

	var assignDefault = assignDefaultExpression.Compile();
	assignDefault(order);

	expected.Sections.Single(s => s.Id == "ordering-party").Status = "incomplete";

	context.Order = order;

	using var controller = context.OrdersController;

	var result = await controller.GetOrderSummaryAsync(orderId) as OkObjectResult;
	var orderSummary = result.Value as OrderSummaryModel;
	orderSummary.Should().BeEquivalentTo(expected);
}
```
However, I'm normally a proponent for keeping tests as simple as possible and my concern with this approach is that it's a bit 'too clever' for a test, so I'm wondering if `TestCaseSource` as suggested by @sam-roberts-bjss might be a better fit...

File: src/NHSD.BuyingCatalogue.Ordering.Domain/OrderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NHSD.BuyingCatalogue.Ordering.Domain
+{
+    public static class OrderExtensions
```
[the remark this answers is not in the document]
Comment (SimonLLawrence) [reply src/NHSD.BuyingCatalogue.Ordering.Domain/OrderExtensions.cs]: Removed

File: src/NHSD.BuyingCatalogue.Ordering.Domain/OrderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NHSD.BuyingCatalogue.Ordering.Domain
+{
+    public static class OrderExtensions
+    {
+        public static bool IsContactComplete(this Contact contact)
```
[the remark this answers is not in the document]
Comment (SimonLLawrence) [reply src/NHSD.BuyingCatalogue.Ordering.Domain/OrderExtensions.cs]: removed

File: src/NHSD.BuyingCatalogue.Ordering.Domain/OrderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+{
+    public static class OrderExtensions
+    {
+        public static bool IsContactComplete(this Contact contact)
+        {
+            if (contact == null)
+            {
+                return false;
+            }
+            else
```
[the remark this answers is not in the document]
Comment (SimonLLawrence) [reply src/NHSD.BuyingCatalogue.Ordering.Domain/OrderExtensions.cs]: removed

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs (hunk 1 above)
Comment (SimonLLawrence) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs]: Switched to using Sams TestSource and removed individual field tests, Completion logic now only use OrganisationContact !=null to decide if OrderParty is complete.

**Comment by sam-roberts-bjss** [comment]
I forgot to check for any integration tests. Have these been done as well?

**Pull request merged**